build-logic/src/main/kotlin/com/ibm/wala/gradle/JavaCompileUsingEcj.kt

```diff
@@ -69,10 +70,19 @@ abstract class JavaCompileUsingEcj : JavaCompile() {
 
   @TaskAction
   protected override fun compile(inputs: InputChanges) {
+    val testArgs = options.allCompilerArgs
+    val f = kotlin.io.path.createTempFile("kotlinTemp", "tmp")
+    f.toFile().deleteOnExit()
+    f.toFile().printWriter().use { writer ->
+      for (testArg in testArgs) {
+        writer.print(testArg + " ")
+      }
+    }
+
     execOperations.javaexec {
       classpath(ecjJar)
       executable(javaLauncherPath.get())
-      args(options.allCompilerArgs)
+      args("@" + f)
```

Comment:
Please consider the following further improvements to

* use a better name than `f` and
* prefer string templates over concatenations.

```suggestion
      args("@$moreArgsFile")
```